tox.ini

```diff
@@ -41,6 +42,14 @@ deps =
 commands =
     {envpython} -m sphinx -aWEqb html docs/source docs/build
 
+[testenv:black]
```

Comment:
Is this setup purely for test running/setup in a CI environment? It seems like there might be some conflict between what we have in `pyproject.toml` and what gets installed here.

Reply:
I'll have to double-check. tox -- like pre-commit -- sets up isolated virtual environments with the listed dependencies. However, while I explicitly listed the dependence on `black` I also said to use `black` from within the existing environment so I need to double-check the outcome here, and I enabled the `skip_install` option so I need to double-check the outcome.

I'm pretty sure it's not setting up a virtual environment for these yet but I'll double-check and get back to you tomorrow morning. Thanks for noticing this and bringing it up!

Reply:
It's very definitely running from within a tox virtual environment but I've made that more explicit by using syntax like this:

```
{envpython} -m black ...
{envpython} -m flake8 ...
{envpython} -m isort ...
{envpython} -m mypy ...
```

I also manually uninstalled black/flake8/isort/mypy and confirmed that tox was still able to successfully run the lint tests.

The end result is that the tests are independent of the developers' current paths, environments, and setup.